### ISSUE TITLE
Use keras .fit() method to train adversarial models

### ISSUE THREAD

**Issue by kristofgiber** [tensorflow/tensorflow#32328]
TF2.0.0-RC0 keras question:

Because GANs and adversarial models require training two models with different optimizers simultaneously, I don't think this can be achieved currently without a custom training loop, applying only the model.fit() methods. Is this feature already available because I haven't found it in the documentations? To clarify, the feature I would need is to create the generative and the discriminator models, compile the generative model with optimizer_A, compile the discriminator model with optimizer_B, now train them simultaneously but alternately, eg. one batch with discriminator then one batch with generative (using the output of the discriminator model of course for the loss), and do this without having to write a custom training loop. This is the way to go with GANs, but the examples I see all use custom train steps. The problem with custom train steps have limited access to many of the keras features such as saving models, monitoring, etc.

**Comment by ravikyram** [comment]
Do you have any use case that requires the feature you are interested in? Please feel free to submit a PR if you have use cases that supports that feature.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jvishnuvardhan** [comment]
@kristofgiber As mentioned by @ravikyram please provide any use case for this kind of feature? Thanks!

**Comment by kristofgiber** [comment]
Hi! Sorry for the delay. My use case was training GANs, any GANs, but I guess if this wasn't clear from my original post then maybe I don't have a strong case and I may be overlooking something trivial. Unfortunately I can't recall the details as Ive moved on. Thank you for your feedback

**Comment by pavithrasv** [comment]
Adding @omalleyt12 who is working on refactoring code to make it easier for users to write custom training loops.

**Comment by omalleyt12** [comment]
@kristofgiber Thanks for the issue!

We've been working on enabling this very use case. In the latest `tf-nightly`, you can do things like this:

```python
class MyModel(tf.keras.Model):
  def train_step(self, data):
    x, y = data
    with tf.GradientTape() as tape:
      ...
    self.optimizer1.apply_gradients(...)
    self.optimizer2.apply_gradients(...)

  def compile(self, optimizer1, optimizer2, ...):
    self.optimizer1 = optimizer1
    self.optimizer2 = optimizer2
    ...
    self._is_compiled = True
```

This allow users to provide custom training logic in only a few lines of code, while still supporting all data formats, distribution strategies, Callbacks, etc that Model.fit supports

We'll have detailed guides on how to use these new features soon!
